packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx

```diff
@@ -148,7 +148,7 @@ export const DiagramRenderer = ({ diagramRefreshedEventPayload }: DiagramRendere
     }
   }, [diagramRefreshedEventPayload, diagramDescription]);
 
-  const { updateSelectionOnNodesChange, updateSelectionOnEdgesChange } = useDiagramSelection();
+  useDiagramSelection();
```

Comment:
It seems strange to call `useDiagramSelection();` without using it after.
Are you sure about that?

Reply:
Yes, I need to declare it  to trigger the useEffect and the useOnSelectionChange

Reply:
So, is it still relevant to extract the code in a separate file?
Is the name of the file should still start with `use****`?

Reply:
It doesn't seem to be a bad practice to have a hook that returns nothing.
But you're right, we could put the code directly into `DiagramRenderer`, but that would reduce the readability of the code, since `DiagramRenderer` is already a big class with a lot of code.
Even if there is no return data, `useDiagramSelection` is still a custom hook, it should keep a name like `use****`

Reply:
Ok thank you for the clarification